src/averageAges.js

```diff
@@ -15,11 +15,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let mens = people.filter(person => person.sex === 'm');
+
+  if (century !== undefined) {
+    mens = mens.filter(person => Math.ceil(person.died / 100) === century);
+  }
+
+  const totalAge = mens.reduce((sum, person) => sum
+    + (person.died - person.born), 0);
+  const averageAge = parseFloat((totalAge / mens.length).toFixed(2));
```

Comment:
toFixed(2) method rounds the number to keep only two decimals. But parseFloat() could introduce floating point precision issue. It's better to use Number() instead of parseFloat().